sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyCurveName.cs

```diff
@@ -9,15 +9,49 @@ namespace Azure.Security.KeyVault.Keys
     /// <summary>
     /// Elliptic Curve Cryptography (ECC) curve names.
     /// </summary>
-    [Flags]
     public enum KeyCurveName : uint
     {
         P256 = 0x0001,
         P384 = 0x0002,
         P521 = 0x0004,
         P256K = 0x0008,
         Other = 0x0010,
+    }
+
+    public static class KeyCurveNameExtensions
+    {
+        public static KeyCurveName ParseFromString(string value)
+        {
+            switch (value)
+            {
+                case "P-256":
+                    return KeyCurveName.P256;
+                case "P-256K":
+                    return KeyCurveName.P256K;
+                case "P-384":
+                    return KeyCurveName.P384;
+                case "P-521":
+                    return KeyCurveName.P521;
+                default:
+                    return KeyCurveName.Other;
+            }
+        }
 
-        All = uint.MaxValue
+        public static string AsString(KeyCurveName curve)
```

Comment:
FWIW: I'd call these something like `ToWireFormat` and `FromWireFormat`.  Though I'd hope for a better word than "Wire" if it's actually public.

It's also weird (to me) that they're on a class named Extensions yet neither are an extension method.

Reply:
Good point. 
We are having a  conversation to decide if we want to keep this as enum or if we want to change it to a class with static fields. 
While in the process of that comparison I will modify names accordingly.